src/components/Modal.astro

```diff
@@ -0,0 +1,89 @@
+---
```

Comment:
Refactor (blocking): Since the modal is part of the client-side interactivity, and we have React in the project, this would make more sense as a React component.

You can literally copy all that markup into a Modal.tsx component and pass it in a `roomId` and `username` as props.

You can still use this logic in the React component, but reworked a bit.

```javascript
  const dialog = document.querySelector("dialog");
  const returnHomeBtn = document.getElementById("return-home-btn");
  const leaveRoomBtn = document.getElementById("leave-room-btn");
  const stayInRoomBtn = document.getElementById("stay-in-room-btn");

  const closeModal = () => dialog?.close();

  const returnHome = () => {
    window.location.href = "/";
    closeModal();
  };

  leaveRoomBtn?.addEventListener("click", () => {
    dialog?.showModal();
  });

  returnHomeBtn?.addEventListener("click", returnHome);

  stayInRoomBtn?.addEventListener("click", closeModal);
```

For the click events, they can use `onClick`s in the JSX which means you don't need

```javascript
  const returnHomeBtn = document.getElementById("return-home-btn");
  const leaveRoomBtn = document.getElementById("leave-room-btn");
```

For the `showing/closing of the modal, `useState` can be used.

Reply:
I originally tried using a react component but ran into issues around the component showing immediately because of the `client:load`

So I had tried conditionally showing the JSX like this so the modal only shows when the button is clicked but ran into issues with the modal buttons not working. 

```js
{isModalShowing &&  <dialog></dialog>}
```
I will need to spend some more time, looking into how to get it to work so it shows at the correct times and the buttons still work.

Reply:
Also, since we are converting this to a react component, I wonder if refs would be better here 

```js
  const dialog = document.querySelector("dialog");
  const stayInRoomBtn = document.getElementById("stay-in-room-btn");
```